pkg/deploy/api/validation/validation.go

```diff
@@ -50,6 +52,56 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	return allErrs
 }
 
+func resetEmptyImageReferences(changedContainers []*kapi.Container) {
+	if changedContainers != nil && len(changedContainers) > 0 {
+		for _, container := range changedContainers {
+			container.Image = ""
+		}
+	}
+}
+
+func handleEmptyImageReferences(template *kapi.PodTemplateSpec, triggers []deployapi.DeploymentTriggerPolicy) []*kapi.Container {
+	// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
+	// to modify this method's local copy (a pointer was NOT used for the parameter) by setting the field to a non-empty value to
+	// work around the k8s validation as our ICT will supply the image field value
+	containerEmptyImageInICT := make(map[string]bool)
+	for _, container := range template.Spec.Containers {
+		if len(container.Image) == 0 {
+			containerEmptyImageInICT[container.Name] = false
+		}
+	}
+	needToChangeImageField := false
+	if len(containerEmptyImageInICT) > 0 {
+		for _, trigger := range triggers {
+			// note, the validateTrigger call above will add an error if ImageChangeParams is nil, but
+			// we can still fall down this path so account for it being nil
+			if trigger.Type != deployapi.DeploymentTriggerOnImageChange || trigger.ImageChangeParams == nil {
+				continue
+			}
+
+			for _, container := range trigger.ImageChangeParams.ContainerNames {
+				needToChangeImageField = true
+				containerEmptyImageInICT[container] = true
+			}
+		}
+	}
+
+	if needToChangeImageField {
+		changedContainers := make([]*kapi.Container, 0)
+		for i, container := range template.Spec.Containers {
+			// only update containers listed in the icp
+			match, ok := containerEmptyImageInICT[container.Name]
+			if match && ok {
+				changedContainers = append(changedContainers, &(template.Spec.Containers[i]))
+				template.Spec.Containers[i].Image = "unset"
+			}
+		}
```

Comment:
OK, removed the array return, did not add a map return, and inlined the saving of the original names and restoring of them in the validate method itself.